lib/rubocop/config_loader.rb

```diff
@@ -58,7 +58,11 @@ def merge(base_hash, derived_hash)
 
       def base_configs(path, inherit_from)
         configs = Array(inherit_from).map do |f|
-          f = File.join(File.dirname(path), f) unless f.start_with?('/')
+          if f.include?('~')
+            f = File.expand_path(f)
+          else
+            f = File.join(File.dirname(path), f) unless f.start_with?('/')
+          end
```

Comment:
Replacing the conditional with
```ruby
f = File.expand_path(f, File.dirname(path))
```
seems to pass the tests.

Turns out `path` is the path to a file, and `File.expand_path` only handles directories as its second argument. This would result in errors like the following.
```
  14) RuboCop::ConfigLoader.load_file returns a configuration loaded from the passed path
     Failure/Error: described_class.load_file(configuration_path)
     Errno::ENOTDIR:
       Not a directory @ rb_sysopen - /tmp/rubocop/config/default.yml/enabled.yml
```